### PR TITLE
Fix/7236 menu dynamic type 

### PR DESCRIPTION
**Pull request by etoledom**
Fixes part of #7236

This PR adds support for dynamic type in the Menus section.

artboard

**Design**:
The font size of the cells highlighted changed from 14 to 15 (for a simpler implementation of dynamic type)

artboard_2

@iamthomasbishop Could you please check if that's fine?

To test:
- Go to the Blog -> Menus section
- Test all the functionality of that section with different font sizes

### CODE REVIEW

**Comment by iamthomasbishop** [comment]
I think this looks good - bumping the size up to 15 is probably a good call, especially in the context of adding flexibility for dynamic type.

File: WordPress/Classes/ViewRelated/Menus/MenuDetailsViewController.m [hunk 1]
```diff
@@ -43,15 +44,37 @@ - (void)viewDidLoad
     [self setupTextFieldDesignViews];
     [self setupDoneButton];
     [self setupTrashButton];
+    [self setupObservers];
 
     [self updateTextFieldDesignIconPositioning];
 }
 
+-(void)dealloc
```
Comment (ctarda) [inline WordPress/Classes/ViewRelated/Menus/MenuDetailsViewController.m]: If I recall correctly, starting with iOS 9, it is not necessary to remove observer on dealloc. I am not saying we should remove this code, just that in theory it is not necessary 😉

File: WordPress/Classes/ViewRelated/Menus/MenuDetailsViewController.m [hunk 2]
```diff
@@ -43,15 +44,37 @@ - (void)viewDidLoad
...
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+}
+
+- (void)contentSizeCategoryDidChange
+{
+    [self updateTextFieldFont];
+    [self updateTextFieldDesignIconPositioning];
+}
+
+- (void)setupObservers
```
Comment (ctarda) [inline WordPress/Classes/ViewRelated/Menus/MenuDetailsViewController.m]: During my latest round of fights with dynamic type, I found that we can achieve the same result (being notified of changes in content size) by means of overriding `traitCollectionDidChange`. In my opinion it makes reacting to changes in text size slightly cleaner, and it does not require observing any notification.

**Comment by etoledom** [comment]
@ctarda Thanks for the tip! I implemented the content size change with `traitCollectionDidChange` and it's definitely more elegant.

Thanks for the tip!

**Comment by ctarda** [comment]
:shipit:

**Pull request merged**